### DISCUSSION TITLE
Can i use this library for usb mass storage Pen drive ?

### DISCUSSION THREAD

**Discussion by asking23** [tobozo/ESP32-USB-Soft-Host#44] [General]
Hi,

I need to use this library with ESP32 WROOM and need to use Pen drive for storing the Data Log files. Can i use this library or need another library ?

**Comment by tobozo** [comment]
Hi,

nope you can't, this library is limited to USB-LS **HID** devices.

Only an ESP32-S2 or ESP32-S3 will be able to act as a pen drive, and you won't need any external library for that as they're natively supporting USB-OTG.

**Reply by asking23** [reply]
I want to write log files to Pen Drive. So Need USB Host Functionality.

**Reply by tobozo** [reply]
You need a full usb HOST with USB-MSC support, it still requires and ESP32-S2 or S3 and ESP32-USB-Soft-Host is useless for that as USB-MSC needs at least USB 2.0. 

Also I don't know of any *host* implementation of USB-MSC so you'll probably have to bake you own. 

I have no idea how to do that, but I'll be interested by the results of your research, feel free to add more feedback to this post 👍
